docs/DeveloperGuide.md

```diff
@@ -1,34 +1,965 @@
+This document is meant for new and current developers of AniChan. It describes the overall architecture design of AniChan, and lays out the current implementation details of our notable features with the rationale and considerations behind each one. It is a living document that would continue to be edited and updated for each major release, and the current edition of this document is intended for the release v2.0.
 
-## Glossary
+<br/>
 
-* *glossary item* - Definition
+## 2. Setting Up
+### 2.1 Setting up the project in your computer
 
-## Instructions for manual testing
+Ensure that you have the following installed: 
+*   JDK 11.
+*   IntelliJ IDE (highly recommended).
 
-{Give instructions on how to do a manual product testing e.g., how to load sample data to be used for testing}
+Firstly, **fork** this repo and **clone** a copy into your computer.
+
+If you plan to use Intellij IDEA: 
+1. **Ensure IntelliJ is configured to use JDK 11**.
+    1. Click on `Configure` > `Structure for New Projects` > `Project Settings` > `Project`, 
+       and ensure the `Project SDK` is using **JDK 11**.
+2. **Import the project as a Gradle project**.
+    1. Click on `Import Project` and locate the `build.gradle` file and select it. Click `OK`
+    2. If asked, choose to `Open as Project` (not `Open as File`).
+    3. Click `OK` to accept the default settings but do ensure that the selected version of `Gradle JVM` matches the JDK being used for the project.
+    4. The import process could take a few minutes to finish.
+3. **Verify the setup**: 
+    1. Run the `seedu.anichan.Main` and try a few commands. 
+    2. You may want to refer to our [User Guide](UserGuide.md) for the list of commands.
+
+----
+
+### 2.2 Before writing code
+
+#### 2.2.1 Configuring the Coding Style
+
+If using IDEA, follow this guide [IDEA: Configuring the code style](https://se-education.org/guides/tutorials/intellijCodeStyle.html) to setup IDEA’s coding style to match ours.
+    
+#### 2.2.2 Set up CI
+    
+There is no set up required as the project comes with a GitHub Actions config files, located in `.github/workflows` folder. When GitHub detects these files, it will run the CI for the project automatically at each push to the master branch or to any PR.
+    
+#### 2.2.3 Learn the Design
+    
+When you are ready to start writing codes, we recommended that you have a look at AniChan's overall design by reading about it at [AniChan's architecture](DeveloperGuide.md#31-architecture).
+    
+<br/>
+
+## 3. Design 
+
+The following section describes the architecture design of **Anichan**. This section starts off by looking at the overall architecture design in a general view, before going into the specific implementation details of the individual features.
+
+<br/>
+
+### 3.1 Architecture
+This section will help provide insight to the general overview of Anichan’s architecture.
+
+<br/>
+
+![Architectural Diagram](images/Architectural-Design.png)
```

Comment:
I suggest you could put main in the middle and connect the rest of the relevant boxes to it rather than the floating arrows which may arise confusion.